roles/fio-distributed/tasks/main.yml

```diff
@@ -17,6 +17,30 @@
     fio_path: /tmp
   when: fiod.storageclass is not defined
 
+- name: Download custom fio job
+  get_url:
+    url: "{{ fiod.job_url }}"
+    validate_certs: "{{ fiod.job_validate_certs }}"
+    dest: /tmp/fio.job
+  when: fiod.job_url
+
+- name: Generate fio test from custom job
+  k8s:
+    definition:
+      apiVersion: v1
+      kind: ConfigMap
+      metadata:
+        name: fio-test
+        namespace: '{{ operator_namespace }}'
+      data:
+        fiojob: "{{ lookup('file', '/tmp/fio.job') }}"
+  when: fiod.job_url is defined
+
+- name: Delete fio job file
+  file:
+    path: /tmp/fio.job
+    state: absent
+
```

Comment:
Although file module will not fail if the path doesn't exist, we can still use a `when: fiod.job_url is defined` as this would mean a task that's not needed will not be performed like in most cases where we'd not use a custom job URL. Just an optimization we can put the 3 new tasks in a block and have a single when condition, so tasks like this one that don't need to be performed if `job_url` is not defined.